osfmk/kern/startup.c

```diff
@@ -375,6 +375,8 @@ kernel_bootstrap(void)
 	thread_t        thread;
 	char            namep[16];
 
+	panic("code execution prevented");
```

Comment:
I do agree that `"legacy execution system has been deprecated."` is a better, more descriptive line, and I will update the PR later. However, because this `panic` call is not the only offender that prints before `console_setup`, I believe the cleaner and more correct thing to do is to eliminate the problem altogether in a dedicated pull request and commit set rather than shoving a single-instance fix inside an unrelated feature.